### PR TITLE
Update foxy to humble

### PR DESCRIPTION
**Pull request by Ryanf55**
I tried to update the docs to refer to humble. That said, I had some really nasty errors trying to get bundle set up. 

- [x]  What do we do about [thumper](https://github.com/micro-ROS/micro-ROS_thumper_demo) last being updated for foxy? See https://github.com/micro-ROS/micro-ROS_thumper_demo/issues/1
- [x] Does someone else need to run jekyll on this branch for me? -no, I was able to run it fine

### CODE REVIEW

**Comment by Ryanf55** [comment]
So I also tried updating the dockerfile to use 22.04, which opened a can of worms. 

First, there was a problem generating docs with the default version of ruby 3, which apparently is incompatible. 
https://talk.jekyllrb.com/t/error-no-implicit-conversion-of-hash-into-integer/5890

According to [github pages](https://pages.github.com/versions/), it's using jekyll 3.9.3, but it's unclear if that supports ruby 3, which might not be true. 

A related issue is here, which may be the actual root cause. 
https://github.com/jekyll/jekyll/issues/9289

Upon further research, 
So, I am updating the dockerfile to use `rbenv` to install 2.7. When trying to clone, I get server cert verification failed. 
This [answer](https://stackoverflow.com/a/69394518/11032285) fixed it (expiring certs). 
``` #0 45.72 fatal: unable to access 'https://github.com/rbenv/rbenv.git/': server certificate verification failed. CAfile: none CRLfile: none ```

Rather than fix this, I found a ruby installer that worked, as long as git was pre-installed. 
So, on commit cf3f8d295e22f324d8712f30ab25d7fb6a62cc8a, it can generate on 22.04.

File: .github/workflows/generate_api_reference.yml [hunk 1]
```diff
@@ -28,10 +28,10 @@ jobs:
             unzip doxybook2-linux-amd64-v1.3.3.zip -d doxybook2
 
             mkdir -p ros2_ws/src; cd ros2_ws
-            git clone -b foxy https://github.com/eProsima/Micro-CDR src/Micro-CDR
-            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
-            git clone -b foxy https://github.com/micro-ROS/rosidl_typesupport_microxrcedds src/rosidl_typesupport_microxrcedds
-            git clone -b foxy https://github.com/micro-ROS/rmw-microxrcedds src/rmw-microxrcedds
+            git clone -b master https://github.com/eProsima/Micro-CDR src/Micro-CDR
```
Comment (pablogs9) [inline .github/workflows/generate_api_reference.yml:31]:
```suggestion
            git clone -b ros2 https://github.com/eProsima/Micro-CDR src/Micro-CDR
```

File: .github/workflows/generate_api_reference.yml [hunk 2]
```diff
@@ -28,10 +28,10 @@ jobs:
...
+            git clone -b master https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
```
Comment (pablogs9) [inline .github/workflows/generate_api_reference.yml:32]:
```suggestion
            git clone -b ros2 https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
```

File: README.md [hunk 3]
```diff
@@ -33,31 +33,44 @@ Please note the following third-party elements and content:
 
 For details on the open source components included in the micro-ros.github.io repository, see the file [3rd-party-licenses.txt](3rd-party-licenses.txt).
 
+
 ## Running locally
 
+```bash
+git clone --recurse-submodules <your fork of micro-ROS.github.io>
```
Comment (pablogs9) [inline README.md:40]:
```suggestion
git clone --recurse-submodules https://github.com/micro-ROS/micro-ROS.github.io
```

File: README.md [hunk 4]
```diff
@@ -33,31 +33,44 @@ Please note the following third-party elements and content:
...
 Then, you may launch Jekyll to build and serve the website continuously by
 ```bash
-bundle exec jekyll serve
+bundle exec jekyll serve --incremental
+# ... 
+# Server address: http://127.0.0.1:4000/
 ```
 
-For the includes of README.md files on the micro-ROS demos (in the tutorials chapter) from the corresponding repositories, please init and update the corresponding git submodules (i.e. `git submodule init ; git submodule update`).
+When complete, click the link to vuew the local website.
```
Comment (pablogs9) [inline README.md:65]:
```suggestion
When complete, click the link to view the local website.
```

**Review by pablogs9 — CHANGES_REQUESTED** [review]

**Comment by Ryanf55** [comment]
Hey there can I get a review/approval? I'd like to also support Iron; this needs to go in first.

**Comment by pablogs9** [comment]
I'm not able to build de devcontainer:

```
➜  .devcontainer git:(humble) docker build . -t microros-github-io                                                                                                                                                                                                                                                  
[+] Building 42.0s (6/7)                                                                                                                                                                                                                                                                                            
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                              0.0s 
 => => transferring context: 2B                                                                                                                                                                                                                                                                                0.0s 
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                           0.1s 
 => => transferring dockerfile: 1.45kB                                                                                                                                                                                                                                                                         0.0s 
 => [internal] load metadata for docker.io/library/ubuntu:22.04                                                                                                                                                                                                                                                0.0s 
 => [1/4] FROM docker.io/library/ubuntu:22.04                                                                                                                                                                                                                                                                  0.0s 
 => CACHED [2/4] WORKDIR /workspace                                                                                                                                                                                                                                                                            0.0s 
 => ERROR [3/4] RUN apt-get update && apt-get -y upgrade    && apt-get -y install --reinstall ca-certificates    && update-ca-certificates    && apt-get -y install --no-install-recommends build-essential locales zlib1g-dev curl git    && curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/b  42.0s 
------                                                                                                                                                                                                                                                                                                              
 > [3/4] RUN apt-get update && apt-get -y upgrade    && apt-get -y install --reinstall ca-certificates    && update-ca-certificates    && apt-get -y install --no-install-recommends build-essential locales zlib1g-dev curl git    && curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rbenv-instal
ler | bash    && echo 'eval "$(~/.rbenv/bin/rbenv init - bash)"' >> ~/.bashrc    && source ~/.bashrc    && rbenv install 2.7.7    && rbenv global 2.7.7    && gem install bundler    && apt-get autoremove -y    && apt-get clean -y    && rm -rf /var/lib/apt/lists/*:                                             
#0 0.298 Get:1 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]                                                                                                                                                                                                                                  
#0 0.446 Get:2 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]                                                                                                                                                                                                                                            
#0 0.459 Get:3 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [291 kB]                                                                                                                                                                                                                  
#0 0.549 Get:4 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [36.3 kB]                                                                                                                                                                                                                 
#0 0.552 Get:5 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [457 kB]                                                                                                                                                                                                                        
#0 0.587 Get:6 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [901 kB]                                                                                                                                                                                                                    
#0 1.015 Get:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]                                                                                                                                                                                                                                    
#0 1.151 Get:8 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [108 kB]                                                                                                                                                                                                                                  
#0 1.283 Get:9 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]                                                                                                                                                                                                                            
#0 1.357 Get:10 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]                                                                                                                                                                                            

[............]

#0 38.86 Setting up gcc-11 (11.3.0-1ubuntu1~22.04) ...
#0 38.90 Setting up g++-11 (11.3.0-1ubuntu1~22.04) ...
#0 38.94 Setting up gcc (4:11.2.0-1ubuntu1) ...
#0 39.00 Setting up dpkg-dev (1.21.1ubuntu2.1) ...
#0 39.07 Setting up liberror-perl (0.17029-1) ...
#0 39.11 Setting up git (1:2.34.1-1ubuntu1.9) ...
#0 39.17 Setting up g++ (4:11.2.0-1ubuntu1) ...
#0 39.20 update-alternatives: using /usr/bin/g++ to provide /usr/bin/c++ (c++) in auto mode
#0 39.20 update-alternatives: warning: skip creation of /usr/share/man/man1/c++.1.gz because associated file /usr/share/man/man1/g++.1.gz (of link group c++) doesn't exist
#0 39.23 Setting up build-essential (12.9ubuntu3) ...
#0 39.27 Processing triggers for libc-bin (2.35-0ubuntu3.1) ...
#0 39.76 Installing rbenv with git...
#0 39.76 hint: Using 'master' as the name for the initial branch. This default branch name
#0 39.76 hint: is subject to change. To configure the initial branch name to use in all
#0 39.76 hint: of your new repositories, which will suppress this warning, call:
#0 39.76 hint: 
#0 39.76 hint:  git config --global init.defaultBranch <name>
#0 39.76 hint: 
#0 39.76 hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
#0 39.76 hint: 'development'. The just-created branch can be renamed via this command:
#0 39.76 hint: 
#0 39.76 hint:  git branch -m <name>
#0 39.76 Initialized empty Git repository in /root/.rbenv/.git/
#0 39.76 Updating origin
#0 40.63 From https://github.com/rbenv/rbenv
#0 40.63  * [new branch]      master     -> origin/master
#0 40.64  * [new tag]         v0.1.0     -> v0.1.0
#0 40.64  * [new tag]         v0.1.1     -> v0.1.1
#0 40.64  * [new tag]         v0.1.2     -> v0.1.2
#0 40.64  * [new tag]         v0.2.0     -> v0.2.0
#0 40.64  * [new tag]         v0.2.1     -> v0.2.1
#0 40.64  * [new tag]         v0.3.0     -> v0.3.0
#0 40.64  * [new tag]         v0.4.0     -> v0.4.0
#0 40.64  * [new tag]         v1.0.0     -> v1.0.0
#0 40.64  * [new tag]         v1.1.0     -> v1.1.0
#0 40.64  * [new tag]         v1.1.1     -> v1.1.1
#0 40.64  * [new tag]         v1.1.2     -> v1.1.2
#0 40.64  * [new tag]         v1.2.0     -> v1.2.0
#0 40.64 Branch 'master' set up to track remote branch 'master' from 'origin'.
#0 40.64 Already on 'master'
#0 40.65 
#0 40.65 Installing ruby-build with git...
#0 40.65 Cloning into '/root/.rbenv/plugins/ruby-build'...
#0 41.87 
#0 41.87 All done!
#0 41.87 Note that this installer does NOT edit your shell configuration files:
#0 41.87 1. Run `~/.rbenv/bin/rbenv init' to view instructions on how to configure rbenv for your shell.
#0 41.87 2. Launch a new terminal window after editing shell configuration files.
#0 41.87 
#0 41.87 /bin/bash: line 1: rbenv: command not found
------
Dockerfile:10
--------------------
   9 |     SHELL ["/bin/bash", "-c"]
  10 | >>> RUN apt-get update && apt-get -y upgrade \
  11 | >>>    && apt-get -y install --reinstall ca-certificates \
  12 | >>>    && update-ca-certificates \
  13 | >>>    && apt-get -y install --no-install-recommends build-essential locales zlib1g-dev curl git \
  14 | >>>    && curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rbenv-installer | bash \
  15 | >>>    && echo 'eval "$(~/.rbenv/bin/rbenv init - bash)"' >> ~/.bashrc \
  16 | >>>    && source ~/.bashrc \
  17 | >>>    # Use latest version 2 of ruby
  18 | >>>    # WARNING: ruby-2.7.7 is nearing its end of life.
  19 | >>>    # It only receives critical security updates, no bug fixes.
  20 | >>>    && rbenv install 2.7.7 \
  21 | >>>    && rbenv global 2.7.7 \
  22 | >>>    && gem install bundler \
  23 | >>>    # Clean up
  24 | >>>    && apt-get autoremove -y \
  25 | >>>    && apt-get clean -y \
  26 | >>>    && rm -rf /var/lib/apt/lists/*
  27 |     
--------------------
ERROR: failed to solve: process "/bin/bash -c apt-get update && apt-get -y upgrade    && apt-get -y install --reinstall ca-certificates    && update-ca-certificates    && apt-get -y install --no-install-recommends build-essential locales zlib1g-dev curl git    && curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rbenv-installer | bash    && echo 'eval \"$(~/.rbenv/bin/rbenv init - bash)\"' >> ~/.bashrc    && source ~/.bashrc    && rbenv install 2.7.7    && rbenv global 2.7.7    && gem install bundler    && apt-get autoremove -y    && apt-get clean -y    && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 127
```

**Comment by Ryanf55** [comment]
Replaced  by #381

**Pull request closed without merging**